tests/test_flexible_unet.py

```diff
@@ -0,0 +1,199 @@
+                            (batch, num_classes) + (input_shape,) * spatial_dim,
+                        ]
+                    )
+    return ret_tests
+
+
+# create list of selected models to speed up redundant tests
+# only test the models B0, B3, B7
+SEL_MODELS = [get_model_names()[i] for i in [0, 3]]
+
+# pretrained=False cases
+# 2D and 3D models are expensive so use selected models
+CASES_2D = make_shape_cases(
+    models=SEL_MODELS,
+    spatial_dims=[2],
+    batches=[1, 4],
+    pretrained=[False],
+    in_channels=3,
+    num_classes=10,
+    norm="instance",
+)
+CASES_3D = make_shape_cases(
+    models=[SEL_MODELS[0]],
+    spatial_dims=[3],
+    batches=[1],
+    pretrained=[False],
+    in_channels=3,
+    num_classes=10,
+    norm="batch",
+)
+
+# varying num_classes and in_channels
+CASES_VARIATIONS = []
+
+# change num_classes test
+# 20 classes
+# 2D
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=SEL_MODELS, spatial_dims=[2], batches=[1], pretrained=[False, True], in_channels=3, num_classes=20
+    )
+)
+# 3D
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=[SEL_MODELS[0]], spatial_dims=[3], batches=[1], pretrained=[False], in_channels=3, num_classes=20
+    )
+)
+
+# change in_channels test
+# 1 channel
+# 2D
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=SEL_MODELS, spatial_dims=[2], batches=[1], pretrained=[False, True], in_channels=1, num_classes=10
+    )
+)
+# 8 channel
+# 2D
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=SEL_MODELS, spatial_dims=[2], batches=[1], pretrained=[False, True], in_channels=8, num_classes=10
+    )
+)
+# 3D
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=[SEL_MODELS[0]], spatial_dims=[3], batches=[1], pretrained=[False], in_channels=1, num_classes=10
+    )
+)
+
+# change input shape test
+# 96
+# 2D 96x96 input
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=SEL_MODELS,
+        spatial_dims=[2],
+        batches=[1],
+        pretrained=[False, True],
+        in_channels=3,
+        num_classes=10,
+        input_shape=96,
+    )
+)
+# 2D 64x64 input
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=SEL_MODELS,
+        spatial_dims=[2],
+        batches=[1],
+        pretrained=[False, True],
+        in_channels=3,
+        num_classes=10,
+        input_shape=64,
+    )
+)
+
+# 3D 32x32x32 input
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=SEL_MODELS,
+        spatial_dims=[2],
+        batches=[1],
+        pretrained=[False],
+        in_channels=3,
+        num_classes=10,
+        input_shape=32,
+    )
+)
+
+# 3D 64x64x64 input
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=SEL_MODELS,
+        spatial_dims=[2],
+        batches=[1],
+        pretrained=[False],
+        in_channels=3,
+        num_classes=10,
+        input_shape=64,
+    )
+)
+
+
+class TestFLEXIBLEUNET(unittest.TestCase):
```

Comment:
HI @binliunls , this PR looks good to me, could you add one more test case which tests the pretrained weights for 2d input?
You can refer to `test_kitty_pretrained` in `test_efficientnet.py`

Reply:
Hi @yiheng-wang-nv ,
 I have added the weight check to the unitest.
Thanks.